common/json_tok.h

```diff
@@ -172,4 +172,24 @@ struct command_result *param_psbt(struct command *cmd,
 				  const char *buffer,
 				  const jsmntok_t *tok,
 				  struct wally_psbt **psbt);
+
+/**
+ * Parse an array of strings into a tal array.
+ */
+struct command_result *param_string_arr(struct command *cmd, const char *name,
+					const char *buffer,
+					const jsmntok_t *tok,
+					const char ***strings);
+
+/**
+ * Parse a list of `txid:output` outpoint strings.
+ *
+ * The elements are kept as strings in case we want to forward them,
+ * however they are verified to be syntactically valid.
+ */
+struct command_result *param_string_outpoint_arr(struct command *cmd,
+						 const char *name,
+						 const char *buffer,
+						 const jsmntok_t *tok,
+						 const char ***strings);
```

Comment:
Despite the inefficiency, I prefer converting rather than leaving strings.  Maybe we need a struct outpoint as a separate patch though.

Reply:
I was unsure whether the added effort and conversions would cause some pushback, but generally I agree that we should always verify as much as possible by doing all conversions, even when doing passthru of arguments (which caused this issue in the first place).